Docs: Add admin docs page for the Outbound request log (#45810)
sourcegraph/sourcegraph: doc/admin/observability/outbound-request-log.md

# Outbound request log

There's a page in the web UI where instance admins can see all outbound requests generated by an instance. The page makes it easy to quickly see how Sourcegraph interacts with other systems, such as code hosts, without manually digging into logs, and is especially useful when debugging.

![UI screenshot](../img/outbound-request-log.png)

## Prerequisites

This document assumes you are a [site administrator](../index.md).

## Enabling the logs

This feature is off by default. You can enable it by setting `outboundRequestLogLimit` to a non-zero value in the site config. The recommended value is `50`.

## Privacy

HTTP headers that are considered sensitive (such as `Authorization`) are redacted.
The logs are stored in memory in Redis and are automatically cleared on an instance restart or when you disable the feature.

> ## Logged data

Reviewer: Perhaps you want to clarify that these are not going in the logs, and that we're talking about what's captured and stored instead.

Author: Great point, thanks, JH!

# Outbound request log

> There's a page in the web UI where instance admins can see all outbound requests generated by an instance. The page makes it easy to quickly see how Sourcegraph interacts with other systems, such as code hosts, without manually digging into logs, and is especially useful when debugging.

Reviewer 2: ```suggestion
On the **Outbound requests** page site-admins can see all outbound requests generated by an instance. The page makes it easy to quickly see how Sourcegraph interacts with other systems, such as code hosts, without manually digging into logs, and is especially useful when debugging.
```

Let's name the page directly :) "There is a page" is confusing as the first sentence. Also: we call them site-admins (or "site admin", see: https://handbook.sourcegraph.com/company-info-and-process/communication/content_guidelines/terminology_guidelines/#term-usage)

![UI screenshot](../img/outbound-request-log.png)

## Prerequisites

> This document assumes you are a [site administrator](../index.md).

Reviewer 2: See above: site admin.

Author: Thanks. I copied this from another section of the docs. I've fixed this at several places in the docs now.

## Enabling the logs

> This feature is off by default. You can enable it by setting `outboundRequestLogLimit` to a non-zero value in the site config. The recommended value is `50`.

Reviewer 2: Can you make `site config` a link to the docs page describing site config? See other examples: https://sourcegraph.sourcegraph.com/search?q=site+config+file%3Adoc&groupBy=repo

Author: That's neat; I didn't know about this possibility.

You can later disable it by setting `outboundRequestLogLimit` to `0`, or by removing the setting entirely.

## Privacy

HTTP headers that may be sensitive (such as `Authorization`) are being redacted before saving. Sensitive information is not stored in the logs.

> Logs are stored in memory in Redis, and are automatically cleared on a Redis instance restart or when you disable the feature.

Reviewer 2: I think that depends on how Redis is configured, right? If they use a single Redis instance configured to be used as cache and store, it won't.

Author: Thanks, modified it to "Logs are stored in memory in Redis with an expiration of seven days. They are automatically cleared when you disable the feature."

## Logged data

This data is logged and displayed by the feature:

- The date and time when the request was sent
- The URL the request was sent to
- The HTTP status code of the response
- The duration of the request
- A stack trace to the code lines the request was sent by
- Any errors
- Request headers (with sensitive fields redacted)
- Response headers (with sensitive fields redacted)
- Request body

## Copying the cURL command

To recreate a request for debugging, you can copy the cURL command for any request by clicking the "Copy curl" button. Then you can paste the command into your terminal and run it, or paste it to something like [Postman](https://www.postman.com/) for further debugging.

> Keep in mind that some headers might be redacted (you'll see the word "REDACTED" in their place, in which case you'll need to add the missing pieces manually.

Reviewer 2: parser error: expected ) got newline instead

Author: Ah, great catch; I'm grateful for this one, none of my tools warned me about this mistake.
